legal-api/src/legal_api/services/business_details_version.py

```diff
@@ -196,215 +192,145 @@ def get_company_details_revision(filing_id, legal_entity_id) -> dict:
+            .order_by(legal_entity_version.change_filing_id).first()
+        return le_revision
 
     @staticmethod
     def get_office_revision(filing, legal_entity_id) -> dict:
         """Consolidates all office changes upto the given transaction id."""
         filing_id = filing.id
         offices_json = {}
-        # address_version = version_class(Address)
-        # offices_version = version_class(Office)
-
-        # offices = db.session.query(offices_version) \
-        #     .filter(offices_version.transaction_id <= transaction_id) \
-        #     .filter(offices_version.operation_type != 2) \
-        #     .filter(offices_version.legal_entity_id == legal_entity_id) \
-        #     .filter(or_(offices_version.end_transaction_id == None,  # pylint: disable=singleton-comparison # noqa: E711,E501;
-        #                 offices_version.end_transaction_id > transaction_id)) \
-        #     .order_by(offices_version.transaction_id).all()
-
-        # for office in offices:
-        #     offices_json[office.office_type] = {}
-        #     addresses_list = db.session.query(address_version) \
-        #         .filter(address_version.transaction_id <= transaction_id) \
-        #         .filter(address_version.operation_type != 2) \
-        #         .filter(address_version.office_id == office.id) \
-        #         .filter(or_(address_version.end_transaction_id == None,  # pylint: disable=singleton-comparison # noqa: E711,E501;
-        #                     address_version.end_transaction_id > transaction_id)) \
-        #         .order_by(address_version.transaction_id).all()
-        #     for address in addresses_list:
-        #         offices_json[office.office_type][f'{address.address_type}Address'] = \
-        #             VersionedBusinessDetailsService.address_revision_json(address)
-
-        # HERE
-        filing_id = filing.id
-        address_history = history_cls(Address)
-        office_history = history_cls(Office)
 
-        offices_current = db.session.query(Office, null().label("changed")) \
-            .filter(Office.change_filing_id <= filing_id) \
+        offices_current = db.session.query(Office, null().label('changed')) \
+            .filter(Office.change_filing_id == filing_id) \
             .filter(Office.legal_entity_id == legal_entity_id) \
-            .filter(Office.deactivated_date is null())
+            .filter(Office.deactivated_date == None)  # pylint: disable=singleton-comparison # noqa: E711,E501;
 
+        office_history = history_cls(Office)
         offices_historical = db.session.query(office_history) \
-            .filter(office_history.change_filing_id <= filing_id) \
+            .filter(office_history.change_filing_id == filing_id) \
             .filter(office_history.legal_entity_id == legal_entity_id)  \
-            .filter(Office.deactivated_date > filing.effective_date)
+            .filter(office_history.deactivated_date > filing.effective_date)
```

Comment:
@thorwolpert since we are filtering `change_filing_id == filing_id` does this condition make sense?

Reply:
i think your suggestion of rolling with `change_filing_id == filing_id` makes a lot of that logic go away.
good catch, and keep simplifying this as much as we can.

Reply:
we still need a field in each history table to track deleted or deactivated as part of a filing